include/z64bgcheck.h

```diff
@@ -52,10 +52,26 @@ typedef struct {
 } CollisionPoly; // size = 0x10
 
 typedef struct {
-    /* 0x00 */ u16 cameraSType;
-    /* 0x02 */ s16 numCameras;
-    /* 0x04 */ Vec3s* camPosData;
-} CamData;
+    /* 0x0 */ u16 setting; // camera setting described by CameraSettingType enum
+    /* 0x2 */ s16 count; // only used when `bgCamData` is a list of points used for crawlspaces
+    /* 0x4 */ s16* bgCamData; // s16 data grouped in threes (ex. Vec3s), is usually of type `BgCamData`, but can be a list of points of type `Vec3s` for crawlspaces
+} BgCamDataHeader; // size = 0x8
+
+typedef BgCamDataHeader CamData; // Todo: Zapd compatibility
+
+// The structure used for all instances of s16 data from `BgCamDataHeader` with the exception of crawlspaces.
+// See `Camera_Subj4` for Vec3s data usage in crawlspaces
+typedef struct {
+    /* 0x00 */ Vec3s pos;
+    /* 0x06 */ Vec3s rot;
+    /* 0x0C */ s16 fov;
+    /* 0x0E */ union {
+                s16 jfifId;
+                s16 timer;
+                s16 flags;
+    };
```

Comment:
minor thing, but styling wise, we probably just want a normal 4-space tab for the union?
```
    /* 0x0E */ union {
        s16 jfifId;
        s16 timer;
        s16 flags;
    };
    ```